contentcuration/contentcuration/management/commands/mark_incomplete.py

```diff
@@ -11,62 +11,64 @@
-        count = self.mark_complete_field(query)
-        logging.info('Marked {} invalid topics (finished in {})'.format(count, time.time() - topicstart))
+        # Mark invalid titles
+        titlestart = time.time()
+        logging.info('Marking blank titles...')
+        count = ContentNode.objects.filter(title='').order_by().update(complete=False)
+        logging.info('Marked {} invalid titles (finished in {})'.format(count, time.time() - titlestart))
+
+        # Mark invalid licenses
+        licensestart = time.time()
+        logging.info('Marking blank licenses...')
+        count = ContentNode.objects.exclude(kind_id=content_kinds.TOPIC).filter(license__isnull=True).order_by().update(complete=False)
+        logging.info('Marked {} invalid licenses (finished in {})'.format(count, time.time() - licensestart))
+
+        licensestart = time.time()
+        logging.info('Marking blank license descriptions...')
+        custom_licenses = list(License.objects.filter(is_custom=True).values_list("pk", flat=True))
+        count = ContentNode.objects.exclude(kind_id=content_kinds.TOPIC)\
+            .filter(license_id__in=custom_licenses).filter(Q(license_description__isnull=True) | Q(license_description=''))\
+            .order_by().update(complete=False)
+        logging.info('Marked {} invalid license descriptions (finished in {})'.format(count, time.time() - licensestart))
 
         # Mark invalid file resources
         resourcestart = time.time()
         logging.info('Marking file resources...')
         i = 0
         count = 0
-        file_check_query = File.objects.filter(preset__supplementary=False, contentnode=OuterRef("id"))
+        file_check_query = File.objects.filter(preset__supplementary=False, contentnode=OuterRef("id")).order_by()
         query = ContentNode.objects \
             .exclude(kind_id=content_kinds.TOPIC) \
             .exclude(kind_id=content_kinds.EXERCISE) \
-            .filter(complete__isnull=True) \
-            .values_list('id', flat=True)
-        node_ids = query[i:i + CHUNKSIZE]
-        while node_ids:
+            .order_by()
+        total = query.count()
+        node_ids = query[i:i + chunksize]
+        while i < total + chunksize:
             nodes = ContentNode.objects.filter(pk__in=node_ids) \
                 .annotate(has_files=Exists(file_check_query)) \
-                .filter(
-                    Q(title='') |
-                    Q(has_files=False) |
-                    Q(license=None) |
-                    (Q(license__is_custom=True) & (Q(license_description=None) | Q(license_description=''))) |
-                    (Q(license__copyright_holder_required=True) & (Q(copyright_holder=None) | Q(copyright_holder='')))
-                ).values_list('id', flat=True)
+                .filter(has_files=False).order_by().values_list('id', flat=True)
```

Comment:
Note the annotation and then filter, although it should use an `WHERE EXISTS` which is great, Django adds a boolean comparison that kills the performance https://code.djangoproject.com/ticket/31453

It's possible we could improve this. I'll take a look